src/vs/workbench/browser/parts/views/treeView.ts

```diff
@@ -348,6 +350,29 @@ abstract class AbstractTreeView extends Disposable implements ITreeView {
 		this._onDidChangeDescription.fire(this._description);
 	}
 
+	private _badge: IViewBadge | undefined;
+	private _badgeActivity: IDisposable | undefined;
+	get badge(): IViewBadge | undefined {
+		return this._badge;
+	}
+
+	set badge(badge: IViewBadge | undefined) {
+
+		if (this._badgeActivity) {
+			this._badgeActivity.dispose();
+			this._badgeActivity = undefined;
+		}
```

Comment:
Here we track the last-set activity for each TreeView (and WebView, below), and whenever a new badge is set, we revoke the old activity and apply the new one.  This does work as expected but I'm not sure if there's a better way to do this.

Reply:
This is the way it's done in other places so it seems fine.